Classes/Slot/ReferenceIndexSlot.php

```diff
@@ -0,0 +1,34 @@
+<?php
+namespace NamelessCoder\AsyncReferenceIndexing\Slot;
+
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+/**
+ * Class ReferenceIndexSlot
+ *
+ */
+class ReferenceIndexSlot
+{
+
+    /**
+     * Exclude tables from ReferenceIndex which cannot contain a reference
+     *
+     * @param string $tableName Name of the table
+     * @param bool &$excludeTable Reference to a boolean whether to exclude the table from ReferenceIndex or not
+     */
+    public function shouldExcludeTableFromReferenceIndex($tableName, &$excludeTable)
+    {
```

Comment:
I would add an early return here if `$excludeTable` is already true. We don't want to override this decision if another signal listener has made it already.